test/com/google/enterprise/adaptor/prebuilt/DateFilterTest.java

```diff
@@ -0,0 +1,465 @@
+    config.put("key", "lastModified");
+    config.put("date", "1977-06-18");
+    config.put("days", "365");
+    DateFilter transform = DateFilter.create(config);
+  }
+
+  @Test
+  public void testCreate_DateDoesNotMatchFormat() {
+    thrown.expect(IllegalArgumentException.class);
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("format", "yyyy-MM-dd");
+    config.put("date", "6/18/1977");
+    DateFilter transform = DateFilter.create(config);
+  }
+
+  @Test
+  public void testCreate_DateInTheFuture() {
+    thrown.expect(IllegalArgumentException.class);
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("date", "2525-06-18");
+    DateFilter transform = DateFilter.create(config);
+  }
+
+  @Test
+  public void testCreate_DaysInTheFuture() {
+    thrown.expect(IllegalArgumentException.class);
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("days", "-18");
+    DateFilter transform = DateFilter.create(config);
+  }
+
+  @Test
+  public void testToString_CustomFormat() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("format", "yyMMddHHmmssZ");
+    config.put("days", "365");
+    MetadataTransform transform = DateFilter.create(config);
+    assertEquals("DateFilter(lastModified, metadata, yyMMddHHmmssZ, 365 days)",
+        transform.toString());
+  }
+
+  @Test
+  public void testToString_DateValueFilter() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("date", "1977-06-18");
+    MetadataTransform transform = DateFilter.create(config);
+    assertEquals("DateFilter(lastModified, metadata, yyyy-MM-dd, 1977-06-18)",
+        transform.toString());
+  }
+
+  @Test
+  public void testToString_KeysetMetadata() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("keyset", "metadata");
+    config.put("days", "365");
+    MetadataTransform transform = DateFilter.create(config);
+    assertEquals("DateFilter(lastModified, metadata, yyyy-MM-dd, 365 days)",
+        transform.toString());
+  }
+
+  @Test
+  public void testToString_KeysetParams() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("keyset", "params");
+    config.put("days", "365");
+    MetadataTransform transform = DateFilter.create(config);
+    assertEquals("DateFilter(lastModified, params, yyyy-MM-dd, 365 days)",
+        transform.toString());
+  }
+
+  @Test
+  public void testCreate_KeysetBogus() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("key", "lastModified");
+    config.put("days", "365");
+    config.put("keyset", "bogus");
```

Comment:
Above and in other CLs, you moved keyset to just after key, but not here or the rest of the examples in this file.